src/pages/index.js

```diff
@@ -104,25 +148,78 @@ function closeModal(modal) {
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = nameInput.value;
-  profileDescription.textContent = descriptionInput.value;
-  closeModal(editModal);
+  setButtonText(submitButton, true, "Saving...");
+
+  api
+    .editUserInfo({
+      name: nameInput.value,
+      about: descriptionInput.value,
+    })
+    .then((user) => {
+      profileName.textContent = user.name;
+      profileDescription.textContent = user.about;
+      editFormElement.reset();
+      resetValidation(
+        editFormElement,
+        [nameInput, descriptionInput],
+        validationConfig
+      );
+      closeModal(editModal);
+      disableButton(editFormElement.querySelector(".modal__submit-button"));
+    })
+    .catch(console.error);
+  setButtonText(submitButton, true, "Save");
```

Comment:
` setButtonText(submitButton, true, "Save");`

*  According to the task, you need to change submit buttons text (`Saving…`) while waiting for the server response. And the user should see it.

You need to **only** return the **default** button text in the `finally` block, so that the button could get the text regardless of whether the request succeeds or fails. 

Otherwise the button changes immediately after submission